Habitica/src/main/java/com/habitrpg/android/habitica/ui/AvatarView.kt

```diff
@@ -433,6 +437,22 @@ class AvatarView : FrameLayout {
         }
     }
 
+    fun setAvatarWithSelectedBackground(avatar: Avatar, backgroundForPurchase: String?) {

```

Comment:
Can we maybe make this more generic? Right now it only supports changing the background. If we decide to also show a preview for things like gear or other customisations, this would work anymore. Maybe extend the existing setAvatar method with an optional parameter `previews` (not sure if that really is the best name, but what I just came up with), that is a Map<LayerType, String>. That way we could replace each layer of the avatar if we want to.

Reply:
Got it got it that makes sense to me - I like previews so I'll be using that 👍

Reply:
The previews code looks good, but I think we don't actually need this method now anymore. Since enow that can be achieved using the previews one.